Improve and apply style guide (Part 1) (#1090)
SAP/cloud-sdk: docs-js/features/connectivity/generic-http-client.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> This guide explains how you can execute arbitrary HTTP requests using the SAP Cloud SDK for JavaScript.

Reviewer: ```suggestion
This guide explains how to execute arbitrary HTTP requests using the SAP Cloud SDK for JavaScript.
```

## When to Use What?

The SAP Cloud SDK for JavaScript can generate clients for [OData](../odata/execute-odata-request.mdx) and [OpenAPI](../openapi/execute-openapi-request).

> If you use OData or OpenAPI, you should use generated clients if possible.

Reviewer: ```suggestion
Use generated clients for OData or OpenAPI services, if possible.
```

[info] Imperative

  - This can either be a <LinkToLatestJsApiDocs slug="interfaces/sap_cloud_sdk_connectivity.Destination" name="destination object" /> which was already fetched or an <LinkToLatestJsApiDocs slug="interfaces/sap_cloud_sdk_connectivity.DestinationFetchOptions" name="object containing a destination name and an optional JWT" />
- The `requestConfig` argument is of type <LinkToLatestJsApiDocs slug="types/sap_cloud_sdk_http_client.HttpRequestConfig" name="HttpRequestConfig" /> or <LinkToLatestJsApiDocs slug="types/sap_cloud_sdk_http_client.HttpRequestConfigWithOrigin" name="HttpRequestConfigWithOrigin" />
- The `httpRequestOptions` argument is of type `HttpRequestOptions`
  - This parameter is optional
  - Can be used to disable [automatic CSRF token fetching](#csrf-token-fetching)

<!-- Prettier doesn't change this, cf https://docusaurus.io/docs/markdown-features/admonitions#usage-with-prettier -->

:::caution

Note that the `destination` and `requestConfig` parameters have an overlap in the keys they accept.
For example, both have a `url` and an `authorization` property.
For keys that exist for both objects, the `requestConfig` takes precedence.

:::

> The `executeHttpRequest()` function returns a promise of <LinkToLatestJsApiDocs slug="interfaces/sap_cloud_sdk_http_client.HttpResponse" name="HttpResponse" />.

Reviewer: ```suggestion
The `executeHttpRequest()` function returns a promise of type<LinkToLatestJsApiDocs slug="interfaces/sap_cloud_sdk_http_client.HttpResponse" name="HttpResponse" />.
```
